my_Enumerable.rb

```diff
@@ -0,0 +1,153 @@
+  module Enumerable
+    def my_each
+      # your code here
+      return to_enum(:double) unless block_given?
+      for i in 0..self.length-1
+        yield(self[i])
+      end
+      return self
+    end
+
+    def my_each_with_index
+      return to_enum(:double) unless block_given?
+      for i in 0..self.length-1
+        yield(self[i],i)
+      end
+      return self
+    end
+
+    def my_select
+      return to_enum(:double) unless block_given?
+      conditionMet = []
+      self.my_each do |item|
+        if yield(item)
+          conditionMet << item
+        end
+      end
+      conditionMet
+    end
+
+    def my_all?(pattern = nil)
+      self.my_each do |item|
+        if block_given?
+          unless yield(item)
+            return false
+          end
+        else
+          if pattern
+            unless pattern === (item)
+              return false
+            end
+          else
+            if item == nil||item == false
+              return false
+            end
+          end
+        end
+      end
+      return true
+    end
+
+    def my_any?(pattern = nil)
+      self.my_each do |item|
+        if block_given?
+          if yield(item)
+            return true
+          end
+        else
+          if pattern
+            if pattern === (item)
+              return true
+            end
+          else
+            if item == true
+              return true
+            end
+          end
+        end
+      end
+      return false
+    end
+
+    def my_none?(pattern = nil)
+      self.my_each do |item|
+        if block_given?
+          if yield(item)
+            return false
+          end
+        else
+          if pattern
+            if pattern === (item)
+              return false
+            end
+          else
+            if item == true
+              return false
+            end
+          end
+        end
+      end
+      return true
+    end
+
+    def my_count(value = nil)
+      counter = 0
+      self.my_each do |item|
+        if value == nil &&  !block_given?
+          counter += 1
+        else
+          if block_given?
+            if yield(item)
+              counter += 1
+            end
+          else
+            if item == value
+              counter += 1
+            end
+          end
+        end
+      end
+      return counter
+    end
+
+    def my_map(proc = nil)
+      return to_enum(:double) if !block_given? && !proc
+      newArray = []
+      self.my_each do |item|
+          if proc
+            newArray << proc.call(item)
+          else
+            newArray << yield(item)
+          end
+      end
+      return newArray
+    end
+
+    def my_inject(initial = nil, operator = nil)
+      if initial.class == Symbol
+        operator = initial
```

Comment:
Your my_inject method is having some odd behavior when tested without arguments or blocks:
```

irb(main):153:0> [1,2,3,4,5].inject(:+)
=> 15
irb(main):154:0> [1,2,3,4,5].my_inject(:+)
=> 12
irb(main):155:0> [10,50,1,2,3].inject(:+)
=> 66
irb(main):156:0> [10,50,1,2,3].my_inject(:+)
=> 66
irb(main):157:0> [1,2,3].inject(:+)
=> 6
irb(main):158:0> [1,2,3].my_inject(:+)
=> 3
irb(main):159:0> [1,2,4].my_inject(:+)
=> 7
irb(main):160:0> [1,2,4,5].inject(:+)
=> 12
irb(main):161:0> [1,2,4,5].my_inject(:+)
=> 12
irb(main):162:0> [1,2,4].my_inject(:+)
=> 7
irb(main):163:0> [1,2,4].inject(:+)
=> 7
irb(main):164:0> [1,2,3].inject(:+)
=> 6
irb(main):165:0> [1,2,3].my_inject(:+)
=> 3
 
```

Reply:
Good spot I figured that last line I added might cause issues... Got it sorted though.